File: bm_db_xam_platform/bm_db_xam_shared/Controller/MainActivityController.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;
using bm_db_xam_android.Helper;
using bm_db_xam_shared.Database;
using bm_db_xam_shared.Model;

namespace bm_db_xam_shared.Controller
{
    public class MainActivityController
    {
        List<Data> DataList;
        static DataDatabase database;

        public MainActivityController()
        {
            database = new DataDatabase(PathHelper.GetLocalPath("db_xam_platform"));
            DataList = new List<Data>();

            for (int i = 0; i < 1000; i++)
            {
                DataList.Add(new Data());
            }
        }

        public async void LoadData()
        {
            var timer = new Stopwatch();
            timer.Start();

            DataList = await database.GetItemsAsync(); ;

            timer.Stop();
            Debug.WriteLine("LoadTime: " + timer.Elapsed);
            Debug.WriteLine("Datensatz:" + DataList.Count);
        }

        public void ResetList()
        {
            DataList = null;
        }

        public async void InsertData()
        {
            var timer = new Stopwatch();
            timer.Start();

            foreach (Data d in DataList)
            {
                await database.SaveItemAsync(d);
            }

            timer.Stop();
            Debug.WriteLine("InsertTime: " + timer.Elapsed);
        }

    }
}

File: bm_db_android/app/src/main/java/steffen/de/bm_db_android/Controller/MainActivityController.kt
package steffen.de.bm_db_android.Controller

import android.app.Application
import kotlinx.coroutines.experimental.awaitAll
import kotlinx.coroutines.experimental.launch
import kotlinx.coroutines.experimental.runBlocking
import steffen.de.bm_db_android.Database.DataDatabase
import steffen.de.bm_db_android.Model.Data
import java.util.*
import kotlin.concurrent.thread
import kotlin.system.measureTimeMillis

class MainActivityController(application: Application){

    private var dataList: MutableList<Data>? = mutableListOf<Data>()
    private val database = DataDatabase.getDatabase(application)

    init {
        for(i in 1..1000){
            dataList!!.add(Data())
        }
    }

    fun loadData() = runBlocking{

        val timer = measureTimeMillis {
            val job = launch{
                dataList = database.dataDao().all
            }
            job.join()
        }

        println("LoadTime:" + timer)
        println("Datansatz:" + dataList!!.count())
    }

    fun resetList(){
        dataList = null;
    }

    fun insertData() = runBlocking{
        val timer = measureTimeMillis {
            val job = launch {
                for (d in dataList!!) {
                    database.dataDao().insert(d)
                }
            }
            job.join()
        }

        println("InsertTime: " + timer)
    }

}
